lib/flat_search_web/live/page_live.ex

```diff
@@ -18,6 +18,73 @@ defmodule FlatSearchWeb.PageLive do
       |> Filters.change_filter(params)
       |> Map.put(:action, :insert)
 
-    {:noreply, assign(socket, changeset: changeset)}
+    params = Enum.map(params, fn {key, value} -> {key, insensitive_string(value)} end)
+
+    {:noreply,
+     assign(socket, changeset: changeset, flats: Flats.get_flats_by(params), filters: params)}
+  end
+
+  @impl true
+  def handle_info({:flat_created, flat}, socket) do
+    case flat_fulfills_filters?(socket.assigns.filters, flat) do
+      true ->
+        {:noreply, assign(socket, flats: [flat] ++ socket.assigns.flats)}
+
+      false ->
+        {:noreply, assign(socket, flats: socket.assigns.flats)}
+    end
+  end
+
+  @impl true
+  def handle_info({:error, reason}, socket) do
+    {:noreply, assign(socket, reason: reason)}
+  end
+
+  defp flat_fulfills_filters?(user_filters, flat_filters) do
```

Comment:
Let's reconsider if this function could be broken down into smaller functions, explaining better what happens inside by naming instead of using comments

Reply:
The book you gave me is the reason of this func :D

Reply:
the LiveView one?